### PR TITLE
Update readonly attributes

### PR DESCRIPTION
**Pull request by dasuni-30**
### Related Issues
- None

### Related PRs
- None

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 1]
```diff
@@ -39,14 +40,27 @@ import {
     Message
 } from "@wso2is/react-components";
 import Axios from "axios";
-import React, { FunctionComponent, ReactElement, useEffect, useMemo, useRef, useState } from "react";
+import { IdentityAppsError } from "modules/core/dist/types/errors/identity-apps-error";
```
Comment (thivi) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:43]: Shall we import it from `@wso2is/core`?

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 2]
```diff
@@ -91,48 +105,88 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
...
+    const { data: validationData } = useValidationConfigData();
+ 
+    /**
+     * Get username configuration.
+     */
+    useEffect(() => {
+        if (validationData) {
+            setUsernameConfig(getUsernameConfiguration(validationData));
+        }
+
```
Comment (thivi) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:129]:
```suggestion
```

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 3]
```diff
@@ -91,48 +105,88 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
...
+    }, [ validationData ]);
+ 
+    /**
+     * Get username configuration.
+     */
+    useEffect(() => {
+
```
Comment (thivi) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:136]:
```suggestion
```

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 4]
```diff
@@ -91,48 +105,88 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
...
+        }
+    }, [ claim, usernameConfig ]);
 
     useEffect(() => {
-        const dialectID = getDialectID();
+        const dialectID: string[] = getDialectID();
 
         if(claim) {
-            const externalClaimRequest = [];
+            const externalClaimRequest: any = [];
```
Comment (thivi) [inline apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx:175]: Shall we avoid using `any` here?

File: apps/console/src/features/users/utils/user-management-utils.ts [hunk 5]
```diff
@@ -40,3 +46,69 @@ export class UserManagementUtils {
...
+
+    const rules: ValidationConfInterface[] = config?.rules;
+
+    if (rules?.length < 1) {
+        return;
+    }
+
+    return {
+        enableValidator: 
+                (getValidationConfig(rules, "AlphanumericValidator", "enable.validator")=="true"
```
Comment (thivi) [inline apps/console/src/features/users/utils/user-management-utils.ts:74]:
```suggestion
                (getValidationConfig(rules, "AlphanumericValidator", "enable. Validator") === "true"
```

File: apps/console/src/features/users/utils/user-management-utils.ts [hunk 6]
```diff
@@ -40,3 +46,69 @@ export class UserManagementUtils {
...
+                || !(getValidationConfig(rules, "EmailFormatValidator", "enable.validator")=="true"))
```
Comment (thivi) [inline apps/console/src/features/users/utils/user-management-utils.ts:75]:
```suggestion
                || !(getValidationConfig(rules, "EmailFormatValidator", "enable. Validator") === "true"))
```

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx (hunk 1 above)
Comment (dasuni-30) [reply apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx]: Addressed by 309ce1299dddd6673c8be1280c6d4230a3b2a27b.

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx [hunk 7]
```diff
@@ -82,7 +96,7 @@ export const EditBasicDetailsLocalClaims: FunctionComponent<EditBasicDetailsLoca
         [ "data-testid" ]: testId
     } = props;
 
-    const dispatch = useDispatch();
+    const dispatch: Dispatch<any>= useDispatch();
```
[the remark this answers is not in the document]
Comment (dasuni-30) [reply apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx]: Addressed by 309ce1299dddd6673c8be1280c6d4230a3b2a27b.

File: apps/console/src/features/users/utils/user-management-utils.ts (hunk 5 above)
Comment (dasuni-30) [reply apps/console/src/features/users/utils/user-management-utils.ts]: Addressed by 309ce1299dddd6673c8be1280c6d4230a3b2a27b.

File: apps/console/src/features/users/utils/user-management-utils.ts (hunk 6 above)
Comment (dasuni-30) [reply apps/console/src/features/users/utils/user-management-utils.ts]: Addressed by 309ce1299dddd6673c8be1280c6d4230a3b2a27b.

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx (hunk 3 above)
Comment (dasuni-30) [reply apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx]: Addressed by 05208a8ce9317ca1e35632b6f4667aa5921ec970.

File: apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx (hunk 4 above)
Comment (dasuni-30) [reply apps/console/src/features/claims/components/edit/local-claim/edit-basic-details-local-claims.tsx]: Addressed with 05049aa1cc24d53927699fa67f76598311183643.

**Review by thivi — APPROVED** [review]

**Pull request merged**